### PR TITLE
Simplify machinery to serialise optional parts of CSS values

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17446)

### CODE REVIEW

**Comment by nox** [comment]
Cc @emilio @SimonSapin for comments.

File: components/style_traits/values.rs [hunk 1]
```diff
@@ -36,25 +36,151 @@ pub trait ToCss {
...
     }
 }
 
+impl<T> ToCss for Option<T>
+where
+    T: ToCss,
+{
+    #[inline]
+    fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: Write {
+        self.as_ref().map_or(Ok(()), |value| value.to_css(dest))
```
Comment (emilio) [inline components/style_traits/values.rs]: I think this would be more readable using a plain old `match`, but w/e.

**Review by emilio — APPROVED** [review]
r=me

**Comment by nox** [comment]
https://treeherder.mozilla.org/#/jobs?repo=try&revision=63d491ad92d291687793466d095563c18e79d076

**Comment by nox** [comment]
https://treeherder.mozilla.org/#/jobs?repo=try&revision=32b6954a2853f2532013ea6ab96c1fde7e496144

**Comment by nox** [comment]
Everything is green.

@bors-servo r=emilio

**Pull request merged**